### PR TITLE
pythonPackages.rdflib: 4.1.2 -> 4.2.2

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -17306,17 +17306,21 @@ in {
 
 
   rdflib = buildPythonPackage (rec {
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: We are moving Python expressions out of `pkgs/top-level/python-packages.nix` into `pkgs/development/python-modules/<module>/default.nix`. 

Please move the expression there, and call it from `pkgs/top-level/python-packages.nix` using `callPackage ../development/python-modules/<package> { };`.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -17306,17 +17306,21 @@ in {
...
+      sha256 = "1j082qr0h7pzr33divq3w4n3k9jpcmbkxnhkw8k3pvq6aqivs0pb";
     };
 
-    # error: invalid command 'test'
+    # tests still use Python 2 syntax:
+    #   File "<...>/rdflib-4.2.1/examples/sparql_query_example.py", line 28
+    #   print row.s
+    #           ^
+    #   SyntaxError: Missing parentheses in call to 'print'. Did you mean print(print row.s)?
     doCheck = false;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: `doCheck = !isPy3k;`

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by bcdarwin** [comment]
@FRidh - fixed.

**Comment by FRidh** [comment]
Pushed 54abfdc667d

**Pull request closed without merging**